app/models/papi_client.rb

```diff
@@ -358,6 +360,64 @@ def get_command_line_options(study_file, action)
     opts
   end
 
+  # set labels for pipeline request/virtual machine
+  #
+  # * *params*
+  #   - +action+ (String, Symbol) => action being executed
+  #   - +study+ (Study) => parent study of file
+  #   - +study_file+ (StudyFile) => File to be ingested/processed
+  #   - +user+ (User) => user requesting action
+  #   - +machine_type+ (String) => GCE machine type
+  #   - +boot_disk_size_gb+ (Integer) => size of boot disk, in GB
+  #
+  # * *returns*
+  #   - (Hash)
+  def job_labels(action:, study:, study_file:, user:, machine_type: DEFAULT_MACHINE_TYPE, boot_disk_size_gb: 300)
+    ingest_version = AdminConfiguration.get_ingest_docker_image_attributes[:tag]
+    {
+      study_accession: sanitize_label(study.accession),
+      user_id: user.id.to_s,
+      filename: sanitize_label(study_file.upload_file_name),
+      action: label_for_action(action),
+      docker_image: sanitize_label(ingest_version),
+      environment: Rails.env.to_s,
+      file_type: sanitize_label(study_file.file_type),
+      machine_type: machine_type,
+      boot_disk_size_gb: sanitize_label(boot_disk_size_gb)
+    }
+  end
+
+  # shorthand label for action
+  #
+  # * *params*
+  #   - +action+ (String) => original action
+  #
+  # * *returns*
+  #   - (String) => label for action, condensing all ingest actions to 'ingest'
+  def label_for_action(action)
+    case action.to_s
+    when /ingest/
+      'ingest_pipeline'
+    when /differential/
+      'differential_expression'
+    when 'render_expression_arrays'
+      'image_pipeline'
```

Comment:
I think we'll want to reserve `image_pipeline` for Image Pipeline proper, i.e. PAPI runs that output images.  If `batch_data_pipeline` doesn't seem fitting, maybe something like `data_cache_pipeline` would work -- anything other than exactly "image_pipeline" seems best.  

```suggestion
      'batch_data_pipeline'
```

Reply:
I think I like `data_cache_pipeline` for this - it matches the job better.

Reply:
[b5c6c07](https://github.com/broadinstitute/single_cell_portal_core/pull/1637/commits/b5c6c07b79446b14091c6ecd277d759362e618fc)